src/Chapter2.hs

```diff
@@ -841,7 +875,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate n l
+  | n < 0 || null l = []
+  | otherwise = subList n (n + length l - 1) (cycle l)
```

Comment:
@vrom911 is there a stack-safe way to do this?

Reply:
Could you please clarify, what do you mean by stack-safe?
If you ask if you can do that in one pass, the answer is yes 🙂 If you mean this, I could elaborate 👌🏼

Reply:
Yes, I'd like a way to know to do this in one pass. Thanks!